### PR TITLE
chore(Data/Nat/Find): move Nat.find to new file

### PR DESCRIPTION
**Pull request by jcommelin**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Nat/Find.lean [hunk 1]
```diff
@@ -0,0 +1,92 @@
+/-
+Copyright (c) 2016 Microsoft Corporation. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Leonardo de Moura, Jeremy Avigad
+-/
+
+import Batteries.WF
+import Mathlib.Init.Data.Nat.Basic
+
+namespace Nat
```
Comment (grunweg) [inline Mathlib/Data/Nat/Find.lean:10]: Can you move this after the module docstring, please? That will also silence the style linter.

File: Mathlib/Data/Nat/Find.lean [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+meaning that you can't just write `find`, even if the `Nat` namespace is open.
+
+The API for `Nat.find` is:
+
+* `Nat.find_spec` is the proof that `Nat.find hp` satisfies `p`.
+* `Nat.find_min` is the proof that if `m < Nat.find hp` then `m` does not satisfy `p`.
+* `Nat.find_min'` is the proof that if `m` does satisfy `p` then `Nat.find hp ≤ m`.
+
+-/
+
```
Comment (grunweg) [inline Mathlib/Data/Nat/Find.lean:27]:
```suggestion

File: Mathlib/Data/Nat/Find.lean [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+      (fun k kn =>
+        ⟨_, fun y r =>
+          match y, r with
+          | _, ⟨rfl, a⟩ => absurd pn (a _ kn)⟩)
+      fun m IH k kn =>
+      ⟨_, fun y r =>
+        match y, r with
+        | _, ⟨rfl, _a⟩ => IH _ (by rw [Nat.add_right_comm]; exact kn)⟩⟩
+
+protected def findX : { n // p n ∧ ∀ m < n, ¬p m } :=
```
Comment (grunweg) [inline Mathlib/Data/Nat/Find.lean:52]: Pre-existing: this definition needs a doc-string. Would you like to write one now?

**Review by grunweg — COMMENTED** [review]
Looks reasonable, thanks for doing this! I have some comments, mostly about pre-existing issues (you're just moving the code).

**Comment by grunweg** [comment]
This needs a `lake exe mk_all` to actually run CI.

**Comment by Ruben-VandeVelde** [comment]
Ref  #11891

**Comment by jcommelin** [comment]
closing in favour of #11891

**Pull request closed without merging**